askama_derive/src/generator.rs

```diff
@@ -1813,6 +1817,7 @@ struct Buffer {
     indent: u8,
     // Whether the output buffer is currently at the start of a line
     start: bool,
+    discard: bool,
```

Comment:
So the previous PR added `discard` in `WritableBuffer`. Remind me why we have both and why we need to discard in both (instead of discarding in one of them only)?

Reply:
One is the actual string where we push content and the other is pending AST that wasn't pushed yet into the string.

So in this case, if we only want to render a specific named block, which is then followed by another one (which we're not interested into), both will get rendered. However, the block we want to render could also be in other blocks, so we still need to go through the whole tree of each present block, but only render the block that was specified.